pkg/resource/comparator.go

```diff
@@ -95,3 +102,102 @@ func (c *comparatorBuilder) BuildAsFunc() (reflect.Type, func(deployed resource.
 	c.Build()
 	return c.comparator.ResourceType, c.comparator.CompFunc
 }
+
+func containAllLabels(deployed resource.KubernetesResource, requested resource.KubernetesResource) bool {
+	deployedLabels := deployed.GetLabels()
+	requestedLabels := requested.GetLabels()
+
+	for key, value := range requestedLabels {
+		if deployedLabels[key] != value {
+			return false
+		}
+	}
+
+	return true
+}
+
+// CreateDeploymentConfigComparator creates a new comparator for DeploymentConfig using Trigger and RollingParams
+func CreateDeploymentConfigComparator() func(deployed resource.KubernetesResource, requested resource.KubernetesResource) bool {
```

Comment:
I assume that you add these functions here to be reused across our CRs? I'm planing to send a PR to `operator-utils` covering this builder. Also, the DC comparator logic isn't already there (on utils)?

Reply:
I guess the DC comparator logic is not generic enough. There are scenarios in which triggers and other details should be handled differently. And the default comparator is not designed to handle such details. So the code is not part of the default comparator in operator-utils. 

The kie-cloud-operator deals with DC in a similar way: https://github.com/kiegroup/kie-cloud-operator/blob/acce7c16d06ea419fa9fdb37c385f721c4b7fa7c/pkg/controller/kieapp/kieapp_controller.go#L214